pkg/volume/glusterfs/glusterfs.go

```diff
@@ -435,6 +389,24 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 			glog.Infof("glusterfs: successfully mounted %s", dir)
 			return nil
 		}
+
+		// Give a try without `auto_unmount mount option, because
+		// it could be that gluster fuse client is older version and
+		// mount.glusterfs is unaware of `auto_unmount`.
+		// Use a mount string without `auto_unmount``
+		autoMountOptions := mountOptions
+		for i := len(autoMountOptions) - 1; i >= 0; i-- {
```

Comment:
I stared at this loop for quite a long time to see what it does and why it needs to process autoMountOptions backwards... IMO, straightforward approach would be more readable and only a bit less efficient (it depends of nr. of auto_unmounts in the array):

```go
autoMountOptions := make([]string, len(mountOptions))
for _, opt := range mountOptions {
    if opt != "auto_unmount" {
         autoMountOptions = append(autoMountOptions, opt)
    }
}
```